sklearn/datasets/base.py

```diff
@@ -229,6 +229,19 @@ def load_files(container_path, description=None, categories=None,
                  DESCR=description)
 
 
+def _mkdirp(d):
```

Comment:
I would put this in https://github.com/scikit-learn/scikit-learn/blob/master/sklearn/utils/fixes.py and use [os.makedirs](https://docs.python.org/3/library/os.html?highlight=makedirs#os.makedirs) for python 3.